cobra/flux_analysis/sampling.py

```diff
@@ -56,6 +56,40 @@
 """
 
 
+def mp_init(obj):
+    """Initialize the multiprocessing pool."""
+    global sampler
+    sampler = obj
+
+
+def shared_np_array(shape, data=None):
+    """Create a new numpy array that resides in shared memory.
+
+    Parameters:
+    ----------
+    shape : tuple of ints
+        The shape of the new array.
+    data : numpy.array
+        Data to copy to the new array. Has to have the same shape.
+    """
+    size = np.prod(shape)
+    array = Array(ctypes.c_double, int(size))
+    np_array = np.frombuffer(array.get_obj())
+    np_array.reshape(shape)
+
+    if data is not None:
+        if len(shape) != len(data.shape):
+            raise ValueError("`data` must have the same dimensions"
+                             "as the created array.")
+        same = all(x == y for x, y in zip(shape, data.shape))
+        if not same:
+            raise ValueError("`data` must have the same shape"
+                             "as the created array.")
+        np_array = data[:]
```

Comment:
Exactly, that was my intention, too. I did some testing and actually either syntax ensures that the values are copied rather than just assigned to the variable name. I typically use the one I suggested in my comment above because `var = x` is usually assignment, so that `var[:] = ...` makes it clear to myself that I want to copy values into `var`. All good :)